src/main/java/com/example/jpa/member/service/MemberService.java

```diff
@@ -18,16 +23,46 @@ public class MemberService {
 
     @Transactional
     public void create(MemberCreateRequest memberCreateRequest) {
+        checkDuplicate(memberCreateRequest);
         memberRepository.save(memberCreateRequest.toEntity());
     }
 
     @Transactional(readOnly = true)
     public MemberResponse findOne(Long id) {
+        return MemberResponse.from(checkExist(id));
+    }
+
+    @Transactional(readOnly = true)
+    public List<MemberResponse> retrieveMembers() {
+        return memberRepository.findAll().stream().map(MemberResponse::from).collect(Collectors.toList());
+    }
+
+    @Transactional
+    public void updateMember(MemberUpdateRequest memberUpdateRequest) {
+        Member targetMember = checkExist(memberUpdateRequest.getId());
+        targetMember.update(memberUpdateRequest);
+    }
+
+    @Transactional
+    public void deleteMember(Long id) {
+        memberRepository.delete(checkExist(id));
+    }
+
+    public Member checkExist(Long id) {
         Member member = memberRepository.findById(id)
                 .orElseThrow(() -> new NoSuchElementException("멤버가 존재하지 않습니다."));
-
-        return MemberResponse.from(member);
+        return member;
     }
 
+    private void checkDuplicate(MemberCreateRequest request) {
+        Member member = memberRepository.findByName(request.getName());
+        if(member != null) {
+            if(member.getPhoneNumber().equals(request.getPhoneNumber())) {
```

Comment:
도메인 객체에게 묻기(`hasSamePhoneNumber`)
if 뎁스 줄이기
```suggestion
        if(member != null && member.hasSamePhoneNumber(request.getPhoneNumber()) {
```

Reply:
다른 피드백과 더해서 Optional로 null에 대한 여부 확인과 hasSamePhoneNumber 메서드를 도메인 객체에 생성해 처리했습니다!